src/puppetlabs/trapperkeeper/authorization/acl.clj

```diff
@@ -79,30 +76,73 @@
   [pattern]
   (-> pattern (str/lower-case) (str/split #"\.") reverse vec))
 
-(def allow-all { :auth-type :allow :type :allow-all :qualifier :exact :length nil :pattern [] })
+(def allow-all {:auth-type :allow
+                :type :allow-all
+                :qualifier :exact
+                :length nil
+                :pattern []})
+
+(def deny-all {:auth-type :deny
+               :type :regex
+               :qualifier :exact
+               :length nil
+               :pattern #"^/.*/$"})
 
 (schema/defn new-domain :- Entry
   "Creates a new ACE for a domain"
   [type :- auth-type
    pattern :- schema/Str]
   (cond
-    ; global allow
-    (= "*" pattern) allow-all
+    (and (= "*" pattern) (= type :allow))
+    allow-all
+
+    ; global deny
+    (and (= "*" pattern) (= type :deny))
+    deny-all
```

Comment:
The subtle differences between how `allow-all` and `deny-all` are handled in the code may look a little strange here.  It seems like maybe the significance of `allow-all` as a first class "type" is basically so that the `ace-compare` logic can weight an `allow-all` ACE as "more important" than any other ACE.  Arguably, it would make sense to weight a `deny-all` ACE as "more important" than any other ACE - with the possible exception of `allow-all`.

That being said, though, I think we might end up simplifying the ACE weighting to just weight any `deny` ACEs above any `allow` ACEs as an outcome of the work to be done for https://tickets.puppetlabs.com/browse/TK-268.  So given that, I think I'm good with just keeping the notion of `deny-all` self-contained here.

Maybe you could inline the `deny-all` map here rather than creating the separate `def` that only gets used once, but that feels cosmetic.  I'm good either way on that part.